apps/remote_control/lib/lexical/remote_control/compilation/dispatch.ex

```diff
@@ -0,0 +1,52 @@
+defmodule Lexical.RemoteControl.Compilation.Dispatch do
+  alias Lexical.RemoteControl
+  alias Lexical.RemoteControl.Api.Messages
+  alias Lexical.RemoteControl.ModuleMappings
+  alias Lexical.RemoteControl.Plugin
+
+  import Messages
+  use GenServer
+
+  def dispatch(module_updated() = message) do
+    GenServer.cast(__MODULE__, {:dispatch, message})
+  end
+
+  def start_link(_) do
+    GenServer.start_link(__MODULE__, [], name: __MODULE__)
+  end
+
+  def init(_) do
+    {:ok, nil}
+  end
+
+  def handle_cast({:dispatch, module_updated() = message}, state) do
+    module_updated(name: module_name, file: filename) = message
+    ModuleMappings.update(module_name, filename)
+    RemoteControl.notify_listener(message)
+    Plugin.on_module_updated(module_name)
+    maybe_report_progress(filename)
+    {:noreply, state}
+  end
+
+  defp maybe_report_progress(file) do
+    if Path.extname(file) == ".ex" do
+      file
+      |> progress_message()
+      |> RemoteControl.notify_listener()
+    end
+  end
+
+  defp progress_message(file) do
+    relative_path = Path.basename(file)
+    message = "compiling: " <> relative_path
+
+    label =
+      if String.starts_with?(relative_path, "deps") do
```

Comment:
`relative_path` become a basename, it will never start with `deps`, so the logic is not correct.

Reply:
I'm ok with this: `deps/.../something/other.ex`

Regarding this prefix `building project compiling`, I think it's best not to add it. This can be easily configured on the client side, whether it's Emacs or neovim.

However, in my understanding, VSCode cannot be configured in this way. Adding such a prefix may cause interference to users since these characters are also quite long: `building project compiling`.

Reply:
> This can be easily configured on the client side, whether it's Emacs or neovim

how? I looked at the config and couldn't find anything in lsp mode that allows this configuration

Reply:
It's also a little strange to point out the length when this pr dramatically reduces it. 

How about <project name> compiling <file>

Reply:
Also, I think you're missing what I'm suggesting. The label would be "building" and the message would be "compiling file" this is fewer characters than what was there before, and if you omit the message as emacs does, it will still make sense